cpp/tests/test_abm_testing_strategy.cpp

```diff
@@ -136,17 +137,21 @@ TEST(TestTestingScheme, initAndRunTestingStrategy)
 
     ScopedMockDistribution<testing::StrictMock<MockDistribution<mio::UniformDistribution<double>>>> mock_uniform_dist;
     EXPECT_CALL(mock_uniform_dist.get_mock(), invoke)
-        .Times(testing::Exactly(2)) //only sampled twice, testing criteria don't apply to third person
-        .WillOnce(testing::Return(0.7))
-        .WillOnce(testing::Return(0.5));
+        .Times(testing::Exactly(6)) //testing criteria don't apply to third person
```

Comment:
That's true. I updated the test with new comments.

Reply:
Thanks. One question: Aren't there three draws of the random distribution per test? One for compliance, one to check if the test shall be performed (random < probability of test) and one for the test result? The second test has only two mocks right now. Since there are no more mocks afterwards, I think you can get back to `testing::Exactly` again if you want to.

Reply:
I think there are 4 draws if the test is positive: 1/Compliance, 2/ Is the test performed?, 3/ The test results and 4/ Does the agent follow isolation? 
I now switch back to using `testing::Exactly` :)

Reply:
Ah, yes. Now it is called 7 times but in `testing::Exactly` it ask for 6 times. Could you check once more? :)